src/QuantumOpticsBase.jl

```diff
@@ -3,6 +3,9 @@ module QuantumOpticsBase
 using SparseArrays, LinearAlgebra, LRUCache, Strided, UnsafeArrays
 import LinearAlgebra: mul!, rmul!
 
+import QuantumInterface: dagger, directsum, ⊕, dm, embed, expect, permutesystems,
+        projector, ptrace, reduced, tensor, ⊗
+
```

Comment:
This is 50% of the changes. Simply moving the namespace for a few functions that would be of use to other libraries, like `QSymbolics` and `QuantumClifford`. The docstrings are mostly still kept in `QuantumOpticsBase` and of course the vast majority of method definitions are still in here too.